internal/services/compute/windows_virtual_machine_resource.go

```diff
@@ -575,84 +625,157 @@ func resourceWindowsVirtualMachineCreate(d *pluginsdk.ResourceData, meta interfa
 		},
 		Tags: tags.Expand(t),
 	}
-
-	if diskControllerType, ok := d.GetOk("disk_controller_type"); ok {
-		params.Properties.StorageProfile.DiskControllerType = pointer.To(virtualmachines.DiskControllerTypes(diskControllerType.(string)))
+	osDiskRaw := d.Get("os_disk").([]interface{})
+	osDisk, err := expandVirtualMachineOSDisk(osDiskRaw, virtualmachines.OperatingSystemTypesWindows)
+	if err != nil {
+		return fmt.Errorf("expanding `os_disk`: %+v", err)
 	}
+	securityEncryptionType := ""
+	if !osDiskIsImported {
+		securityEncryptionType = osDiskRaw[0].(map[string]interface{})["security_encryption_type"].(string)
+		var computerName string
+		if v, ok := d.GetOk("computer_name"); ok && len(v.(string)) > 0 {
+			computerName = v.(string)
+		} else {
+			_, errs := computeValidate.WindowsComputerNameFull(d.Get("name"), "computer_name")
+			if len(errs) > 0 {
+				return fmt.Errorf("unable to assume default computer name %s. Please adjust the %q, or specify an explicit %q", errs[0], "name", "computer_name")
```

Comment:
Same nit as on linux VM, however I do see now that isn't a new err msg, it's only been moved. So fine to ignore these comments

```suggestion
				return fmt.Errorf("unable to assume default computer name %s. Please adjust the `name`, or specify an explicit `computer_name`", errs[0])
```